input/system/azure/logs.go

```diff
@@ -198,30 +205,30 @@ func setupLogTransformer(ctx context.Context, wg *sync.WaitGroup, servers []*sta
 					return
 				}
 
-				// For Flexible Server, attempt direct parsing first (without modifying the message)
-				logLine, ok := logs.ParseLogLineWithPrefix("", in.Properties.Message)
-				if !ok {
-					// For Single Server, the actual Event Hub messages are missing the log_line_prefix and error level, so add them
-					logLineContent := fmt.Sprintf("%s%s:  %s", in.Properties.Prefix, in.Properties.ErrorLevel, in.Properties.Message)
-					logLine, ok = logs.ParseLogLineWithPrefix("", logLineContent)
-					if !ok {
-						logger.PrintError("Can't parse log line: \"%s\"", logLineContent)
-						continue
-					}
+				logLines, azureDbServerName, err := ParseRecordToLogLines(in)
+				if err != nil {
+					logger.PrintError("%s", err)
+					continue
+				}
+				if len(logLines) == 0 {
+					continue
 				}
-				logLine.CollectedAt = time.Now()
-				logLine.UUID = uuid.NewV4()
 
 				// Ignore loglines which are outside our time window (except in test runs)
-				if !logLine.OccurredAt.IsZero() && logLine.OccurredAt.Before(linesNewerThan) && !globalCollectionOpts.TestRun {
+				if !logLines[0].OccurredAt.IsZero() && logLines[0].OccurredAt.Before(linesNewerThan) && !globalCollectionOpts.TestRun {
 					continue
 				}
 
 				foundServer := false
 				for _, server := range servers {
-					if in.LogicalServerName == server.Config.AzureDbServerName {
-						out <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
+					if azureDbServerName == server.Config.AzureDbServerName {
 						foundServer = true
+
+						for _, logLine := range logLines {
+							logLine.CollectedAt = time.Now()
```

Comment:
just curious: if it matters to have the same `collectedAt` across these `logLines` 🤔 it is kinda collected at the same time

Reply:
Its only used for expiring data in the stream handling logic, so it shouldn't matter.

(but good catch to notice that difference!)